src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -977,4 +977,41 @@ void RaftReplDev::cp_flush(CP*) {
 }
 
 void RaftReplDev::cp_cleanup(CP*) {}
+
+void RaftReplDev::gc_repl_reqs() {
+    std::vector< int64_t > expired_keys;
+    m_state_machine->iterate_repl_reqs([this, &expired_keys](auto key, auto rreq) {
```

Comment:
Subjective: Given that we are in same class and we don't take any locks, isn't it easier to avoid another method iterate_repl_reqs and simply access the m_lsn_req_map here and iterate?

Reply:
m_lsn_req_map is not same class as RaftReplDev, right? It is under m_state_machine. 
And RaftReplDev shouldn't access it's member's member directly as general principal, otherwise, if someday m_state_machines changes m_lsn_req_map either to a different name or other details, calls doesn't need to be changed.